### PR TITLE
Schema Block Header unit test, metadata and interaction with OrientDb

### PR DESCRIPTION
**Pull request by nandotorterolo**
## Purpose
Schema Block Header unit test, metadata, and interaction with OrientDb

## Testing

```shell
co.topl.genusLibrary.orientDb.instances.SchemaBlockHeaderSuite:
  + Test Block Header Schema Metadata 3.301s
  + Test Block Header Schema Add Vertex 0.517s

``` 
## Tickets
* subTask BN-935

### CODE REVIEW

File: genus-library/src/test/scala/co/topl/genusLibrary/orientDb/instances/SchemaBlockHeaderSuite.scala [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+import scala.jdk.CollectionConverters._
+
+class SchemaBlockHeaderSuite extends CatsEffectSuite with ScalaCheckEffectSuite with AsyncMockFactory {
+  type F[A] = IO[A]
+  implicit private val logger: Logger[F] = Slf4jLogger.getLoggerFromClass[F](this.getClass)
+
+  test("Block Header Schema Metadata") {
+
+    val res = for {
+      orientGraphFactory <- Resource.pure(new OrientGraphFactory("memory:test"))
```
Comment (SeanCheatham) [inline genus-library/src/test/scala/co/topl/genusLibrary/orientDb/instances/SchemaBlockHeaderSuite.scala:27]: Might help to move this to a test helper at some point if more tests are added

File: genus-library/src/test/scala/co/topl/genusLibrary/orientDb/instances/SchemaBlockHeaderSuite.scala [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+      blockHeader <- ModelGenerators.arbitraryHeader.arbitrary.first.pure[F].toResource
+      vertex <- Sync[F]
+        .blocking(
+          orientGraph
+            .addVertex(s"class:${schema.name}", schema.encode(blockHeader).asJava)
+        )
+        .toResource
+
+      _ <- assertIO(
+        vertex.getProperty[Array[Byte]](schema.properties.filter(_.name == Field.BlockId).head.name).toSeq.pure[F],
```
Comment (SeanCheatham) [inline genus-library/src/test/scala/co/topl/genusLibrary/orientDb/instances/SchemaBlockHeaderSuite.scala:178]: I'm a bit surprised you didn't encounter issues with byte-array comparisons.  Did the `.toSeq` solve that?

**Review by SeanCheatham — APPROVED** [review]

File: genus-library/src/test/scala/co/topl/genusLibrary/orientDb/instances/SchemaBlockHeaderSuite.scala (hunk 1 above)
Comment (nandotorterolo) [reply genus-library/src/test/scala/co/topl/genusLibrary/orientDb/instances/SchemaBlockHeaderSuite.scala:27]: yes, agree, I need to find a way that all database `memory:test`, use the same database, only 1 resource, and drop the database each time each test finish

File: genus-library/src/test/scala/co/topl/genusLibrary/orientDb/instances/SchemaBlockHeaderSuite.scala (hunk 2 above)
Comment (nandotorterolo) [reply genus-library/src/test/scala/co/topl/genusLibrary/orientDb/instances/SchemaBlockHeaderSuite.scala:178]: I did experience issues the first time.  `toSeq` was the suggestion from Munit. +1 to the framework
I will change the name of all test fall,  from `someSuite -> someTest` to be aligned with other files in the repo
thanks for reviewing it

**Pull request merged**